core_v2/sources/domains.move

```diff
@@ -471,31 +499,53 @@ module aptos_names_v2::domains {
             );
         };
 
+        let account_addr = signer::address_of(sign);
+        let target_address_copy = target_address;
+        let result_of_should_set_reverse_lookup_and_target_address = should_set_reverse_lookup_and_target_address_in_register_name(
+            account_addr,
+            target_address,
+            transfer_to_address
+        );
+        let result_of_should_set_target_address = should_set_target_address_in_register_name(
+            subdomain_name,
+            target_address
+        );
+        let transfer_to_address = option::get_with_default(&transfer_to_address, account_addr);
+        let target_address = option::get_with_default(&target_address, account_addr);
+
         // If the token already exists, transfer it to the signer
         // Else, create a new one and transfer it to the signer
-        let account_addr = signer::address_of(sign);
+        // NOTE we don't set target address at this step, we do it after this.
         let token_addr = token_addr_inline(domain_name, subdomain_name);
         if (object::is_object(token_addr)) {
             let record = borrow_global_mut<NameRecord>(token_addr);
             record.expiration_time_sec = name_expiration_time_secs;
             record.target_address = option::none();
-            object::transfer_with_ref(object::generate_linear_transfer_ref(&record.transfer_ref), account_addr);
+            object::transfer_with_ref(object::generate_linear_transfer_ref(&record.transfer_ref), transfer_to_address);
         } else {
             create_token(
-                account_addr,
+                transfer_to_address,
                 domain_name,
                 subdomain_name,
                 name_expiration_time_secs,
             );
         };
 
+        let is_target_address_unset = true;
         let reverse_lookup_result = get_reverse_lookup(account_addr);
         if (option::is_none(&reverse_lookup_result)) {
-            // If the user has no reverse lookup set, set the user's reverse lookup.
-            set_reverse_lookup(sign, subdomain_name, domain_name);
-        } else if (!is_subdomain(subdomain_name)) {
-            // Automatically set the name to point to the sender's address
-            set_target_address_internal(subdomain_name, domain_name, signer::address_of(sign));
+            // If the signer has no reverse lookup set and signer is minting for itself, set the user's reverse lookup and target address.
+            if (result_of_should_set_reverse_lookup_and_target_address) {
```

Comment:
is it possible to separate the logic of reverse look up and set target address in `should_set_reverse_lookup_and_target_address_in_register_name`

I see we also have `should_set_target_address_in_register_name` and it's confusing

Reply:
it's because set reverse lookup will always set target address at the same time, if i break it into 2, i need to break [`set_target_address_and_reverse_lookup`](https://github.com/aptos-labs/aptos-names-contracts/pull/59/files#diff-e92ae37b241c364f4ff602d97ffe2ac754cacda7ca83e29f4e44a4d154977ea9R1041)

Reply:
I do think we should break `set_target_address_and_reverse_lookup` :P 
because those are basically two different things. aren't them? 
you set primary name when the buyer is the owner
you set target address whenever there's a value passing in?